app/assets/templates/file_list.html

```diff
@@ -6,54 +6,75 @@
     Block
   </button>
 
+  <div class="filter">
+    Showing files from
+    <select ng-model="identity_filter.identity.id" ng-options="i.id as i.name for i in identities()">
+      <option value="">--All--</option>
+    </select>
+  </div>
+
   <div ng-if="view_list">
     <table ng-if="hasFiles()">
       <tr>
-        <th><a href="" ng-click="reverse=!reverse;order('name', reverse)">Filename</a></th>
+        <th><a href="" ng-click="reverse = !reverse; order_by = 'name';">Filename</a></th>
         <th>Filetype</th>
-        <th><a href="" ng-click="reverse=!reverse;order('date', reverse)">Identity</a></th>
-        <th><a href="" ng-click="reverse=!reverse;order('date', reverse)">Date</a></th>
+        <th><a href="" ng-click="reverse = !reverse; order_by = 'identity.name';">Identity</a></th>
+        <th><a href="" ng-click="reverse = !reverse; order_by = 'date';">Date</a></th>
         <th>Tags</th>
         <th>&nbsp;</th>
       </tr>
-      <tr ng-repeat="file in files">
-          <td>
-            <a ng-if="file.filetype === 'bruse/url'" ng-href="{{file.url}}" target="_blank">
-              {{file.name}}
+      <!-- this should be added in the future: | IdentityFilter:showIdentities -->
+      <tr ng-repeat="file in files | filter: identity_filter | orderBy : order_by : reverse">
+        <td>
+          <a ng-if="file.url" ng-href="{{file.url}}" target="_blank">
+            <i class="fa fa-external-link"></i> {{file.name}}
+          </a>
+          <span ng-if="!file.url">
+            <a href="" ng-click="download(file.identity.id, file)">
+              <i class="fa fa-download"></i>
             </a>
-            <a ng-if="file.filetype.indexOf('google-apps') > -1" ng-href="{{file.link}}" target="_blank">
-              {{file.name}}
-            </a>
-            <span ng-if="file.filetype != 'bruse/url' && file.filetype.indexOf('google-apps') < 0" ng-click="download(file.identity.id, file)">
```

Comment:
As seen here, links are only to files with 'google-apps' which are the google specific files such as drive text sheets and spreadsheets. I think now it links to every google file regardless of filetype which I don't think we want.